src/lib/yang/data.lua

```diff
@@ -204,9 +204,16 @@ local function struct_parser(keyword, members, ctype)
       return parse1(P)
    end
    local struct_t = ctype and typeof(ctype)
-   local function finish(out)
-      -- FIXME check mandatory values.
-      if struct_t then return struct_t(out) else return out end
+   local function finish(out, leaf)
+     -- FIXME check mandatory values.
```

Comment:
I believe, I've not added any checks for that. There are checks on simple scalar values which is what you see produce the error message in my example but no checks on struct values which still need to be added. This simply allows for omitting optional arguments. Previously it would error when you tried to do `struct_t(nil)` (when `out` was omitted and therefore nil)).

TLDR: yep, i think we still need it :)